### DISCUSSION TITLE
Z-offset in Crazyflie position (LPS)

### DISCUSSION THREAD

**Discussion by laurolg** [bitcraze/discussions#276] [Q&A]
Hello,

I have been having issues with localizing the Crazyflie 2.1 using the Loco Positioning System (LPS). I followed all the steps mentioned on the Bitcraze website for the setup of the Crazyflie 2.1 and the LPS. 

The Crazyflie with its positioning deck are placed inside the area formed by 8 anchors spaced in the room. This issue is that **the z-coordinate of the Crazyflie position is false. There often is an offset of about +15cm**. The x- and y-coordinates are correct.

A few additional information:
- Operating System: Ubuntu 20.04 LTS
- LPS mode: TWR, 8 anchors evenly distributed in the space at least 2m apart and 15cm from walls, ceiling and floor

Any help will be much appreciated.
Thank you very much for your time!

**Comment by krichardsson** [comment]
Hi!

This is a problem that we have seen before and unfortunately we do not fully understand the mechanisms, but my assumption is that it is related to the antenna delay. The antenna delay is the time it takes for the radio waves to propagate through the antenna to the radio and it is in the order the time it takes for light to travel 150 meters. In TWR the antenna delay is removed from the measured distance while it is automatically cancelled out in TDoA. The problem is that the antenna delay depends on the orientation of the antennas, objects close by and so on, and this is not accounted for. You can probably also see that the offest/error is different if you use TWR or TDoA.

One solution that has been used by others is to add a Flow deck or Z-ranger deck to get better data.

Also, try to [mount the anchors with the antenna upwards](https://www.bitcraze.io/documentation/repository/lps-node-firmware/master/user-guides/anchor-setup/), as the radiation pattern is more uniform around this axis.
